File: examples/javascript/text_to_image.mjs
/**
 * Text-to-Image — generate an image from a text prompt.
 *
 * Endpoint: POST /v1/images/generations
 * Model:    nunchaku-qwen-image
 *
 * Usage:
 *   NUNCHAKU_API_KEY=sk-nunchaku-... node text_to_image.mjs
 */

import { writeFileSync } from "fs";

const API_KEY = process.env.NUNCHAKU_API_KEY;
if (!API_KEY) {
  console.error("Set NUNCHAKU_API_KEY environment variable.");
  process.exit(1);
}

const response = await fetch(
  "https://api.nunchaku.dev/v1/images/generations",
  {
    method: "POST",
    headers: {
      Authorization: `Bearer ${API_KEY}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      model: "nunchaku-qwen-image",
      prompt:
        "a red apple on a wooden table, photorealistic, studio lighting",
      n: 1,
      size: "1024x1024",
      tier: "fast",
      num_inference_steps: 28,
      response_format: "b64_json",
      seed: 42,
    }),
  }
);

if (!response.ok) {
  console.error(`Error ${response.status}: ${await response.text()}`);
  process.exit(1);
}

const data = await response.json();
const buf = Buffer.from(data.data[0].b64_json, "base64");
writeFileSync("output_t2i.jpg", buf);
console.log(`Saved output_t2i.jpg (${buf.length.toLocaleString()} bytes)`);

File: examples/python/text_to_image.py
"""Text-to-Image — generate an image from a text prompt.

Endpoint: POST /v1/images/generations
Model:    nunchaku-qwen-image (supports tiers: fast, radically_fast)

Usage:
    export NUNCHAKU_API_KEY="sk-nunchaku-..."
    python text_to_image.py
"""

import base64
import os

import requests

API_KEY = os.environ["NUNCHAKU_API_KEY"]
BASE_URL = "https://api.nunchaku.dev"

response = requests.post(
    f"{BASE_URL}/v1/images/generations",
    headers={
        "Authorization": f"Bearer {API_KEY}",
        "Content-Type": "application/json",
    },
    timeout=180,
    json={
        "model": "nunchaku-qwen-image",
        # nunchaku-qwen-image renders legible in-image text when the prompt
        # (a) quotes the literal string, (b) names a surface to put it on,
        # and (c) specifies a visual style. Meta-instructions like
        # "label this Cat" are unreliable.
        "prompt": (
            "a cat wearing sunglasses on a wooden table, photorealistic, "
            "studio lighting. A small wooden plaque propped on the table "
            "in front of the cat reads \"CAT\" in bold black serif letters."
        ),
        "n": 1,
        "size": "1024x1024",
        "tier": "fast",
        "num_inference_steps": 28,
        "response_format": "b64_json",
        "seed": 4242,
    },
)
response.raise_for_status()

img_bytes = base64.b64decode(response.json()["data"][0]["b64_json"])
with open("output_t2i.jpg", "wb") as f:
    f.write(img_bytes)
print(f"Saved output_t2i.jpg ({len(img_bytes):,} bytes)")
